### PR TITLE
Created the executor package (used by globaldefault)

### PR DESCRIPTION
**Pull request by Naatan**
https://www.pivotaltracker.com/story/show/177552873

### CODE REVIEW

File: test/integration/activate_int_test.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func (suite *ActivateIntegrationTestSuite) activatePython(version string, extraE
 	cp.SendLine("state activate --default something/else")
 	cp.ExpectLongString("Cannot set something/else as the global default project while in an activated state")
 
-	cp.SendLine("state activate --default")
+	cp.SendLine("VERBOSE=true state activate --default")
```
Comment (mdrohmann) [inline test/integration/activate_int_test.go:191]: Can `VERBOSE=true` be removed?

File: test/integration/activate_int_test.go [hunk 2]
```diff
@@ -213,7 +213,7 @@ func (suite *ActivateIntegrationTestSuite) activatePython(version string, extraE
 	cp = ts.SpawnCmdWithOpts(
 		filepath.Join(ts.Dirs.DefaultBin, pythonShim),
 		e2e.WithArgs("-c", "import sys; print(sys.copyright);"),
-		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false", "VERBOSE=true"),
```
Comment (mdrohmann) [inline test/integration/activate_int_test.go:216]: Can `VERBOSE=true` be removed?

File: internal/executor/executor_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	t.Run("Cleanup old executors", func(t *testing.T) {
+		err = fw.Cleanup([]string{exes[1]})
+		require.NoError(t, err, errs.Join(err, ": "))
+
+		files := fileutils.ListDir(fw.BinPath(), false)
+		require.Len(t, files, 1, "Cleanup should only keep one exe")
+		require.Equal(t, filepath.Base(exes[1]), filepath.Base(files[0]), "Cleanup should leave the exe we requested")
+	})
+
+	t.Run("Update doesn't needlessly write", func(t *testing.T) {
```
Comment (mdrohmann) [inline internal/executor/executor_test.go:56]: :+1:

File: internal/globaldefault/default_lin_mac.go [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// +build !windows
+
+package globaldefault
+
+func isOnPATH(binDir string) bool {
+	// todo: https://www.pivotaltracker.com/story/show/177673816
```
Comment (mdrohmann) [inline internal/globaldefault/default_lin_mac.go:6]: It took me a while until I understood what you want to achieve here: You actually want to have a way to figure out whether we have already called `subshell.WriteUserEnv(..., envType)` with that PATH before, correct?

**Review by mdrohmann — CHANGES_REQUESTED** [review]
Looking good and much cleaner now.  I found a few minor issues.

File: internal/executor/executor_windows.go [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package executor
+
+import "strings"
+
+func nameExecutor(exe string) string {
+	exts := strings.Split(strings.ToLower(os.Getenv("PATHEXT")), ";")
+	lowerExe := strings.ToLower(exe)
+	for _, ext := range exts {
+		if strings.HasSuffix(lowerExe, strings.ToLower(ext)) {
+			return exe[0 : len(exe)-ext]
```
Comment (mdrohmann) [inline internal/executor/executor_windows.go:10]:
```suggestion
			return exe[0 : len(exe)-len(ext)]
```

File: internal/globaldefault/default_lin_mac.go (hunk 4 above)
Comment (Naatan) [reply internal/globaldefault/default_lin_mac.go:6]: Correct.

**Review by mdrohmann — APPROVED** [review]

**Pull request merged**